modules/cart/tests/src/Kernel/CartOrderPlacedTest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Drupal\Tests\commerce_cart\Kernel;
+
+use Drupal\commerce_store\StoreCreationTrait;
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\KernelTests\Core\Entity\EntityKernelTestBase;
+
+/**
+ * Tests the unsetting of the cart flag when order is placed.
+ */
+class CartOrderPlacedTest extends EntityKernelTestBase {
```

Comment:
This is resolved. We test creating new cart, which uncovered all those other little bugs.